### PR TITLE
docs: update README and pkg docs

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

File: docs.go [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+//
+// # Examples
+//
+// Gorm: https://github.com/go-gorm/gorm
+//
+//	w, err := mql.Parse("name=alice or name=bob)",User{})
+//	  if err != nil {
+//	  return nil, err
+//	}
+//	err = db.Where(w.Condition, w.Args...).Find(&users).Error
```
Comment (johanbrandhorst) [inline docs.go:16]: Code blocks in Go code should just use a single space indentation: https://go.dev/doc/comment#code.
```suggestion
//  w, err := mql.Parse("name=alice or name=bob)",User{})
//  if err != nil {
//  	return nil, err
//  }
//  err = db.Where(w.Condition, w.Args...).Find(&users).Error
```

I think gofmt should format these automatically though 🤔

Comment (jimlambrt) [reply docs.go:16]: gofmt did format them when I saved my editor.

Comment (johanbrandhorst) [reply docs.go:16]: If nothing else, the alignment on the `if` in this block is off 😁

Comment (johanbrandhorst) [reply docs.go:16]: Sorry, it should be indented by a single tab only, apparently:
> Gofmt indents all lines in a code block by a single tab

Comment (johanbrandhorst) [reply docs.go:16]: In that case I think this should just adjust the if statement indentation

Comment (johanbrandhorst) [reply docs.go:16]: You might want to replace the two spaces in the returns with a tab as well but that's a nit

Comment (johanbrandhorst) [reply docs.go:16]: This if statement is still misaligned

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

File: docs.go [hunk 2]
```diff
@@ -1,31 +1,62 @@
...
+//	q := fmt.Sprintf("select * from users where %s", w.Condition)
+//	rows, err := db.Query(q, w.Args...)
+//
+// go-dbw: https://github.com/hashicorp/go-dbw
+//
+//	w, err := mql.Parse("name=alice or name=bob)",User{})
+//	if err != nil {
+//	  return nil, err
+//	}
+//	err := rw.SearchWhere(ctx, &users, w.C
```
Comment (johanbrandhorst) [inline docs.go:33]: 😬

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**